scripts/common/proxy.sh

```diff
@@ -1,53 +1,94 @@
-
-function proxy_bootstrap() {
+# read_proxy_config_from_env makes sure that both proxy variables (upper
+# and lower case) are set to the same value. for example http_proxy and
+# HTTP_PROXY both must point to the same address. this function sets the
+# following variables ENV_HTTP_PROXY_ADDRESS, ENV_HTTPS_PROXY_ADDRESS,
+# and ENV_NO_PROXY.
+function read_proxy_config_from_env() {
+    # by default https proxy configuration inherits from http proxy config.
+    # we also make sure that we have both environment variables (with upper
+    # and lower case) set to the same value.
     if [ -n "$HTTP_PROXY" ]; then
-        ENV_PROXY_ADDRESS="$HTTP_PROXY"
-        export https_proxy="$HTTP_PROXY"
-        printf "The installer will use the proxy at '%s' (imported from env var 'HTTP_PROXY')\n" "$ENV_PROXY_ADDRESS"
+        ENV_HTTP_PROXY_ADDRESS="$HTTP_PROXY"
+        ENV_HTTPS_PROXY_ADDRESS="$HTTP_PROXY"
+        export http_proxy="$HTTP_PROXY"
     elif [ -n "$http_proxy" ]; then
-        ENV_PROXY_ADDRESS="$http_proxy"
-        export https_proxy="$http_proxy"
-        printf "The installer will use the proxy at '%s' (imported from env var 'http_proxy')\n" "$ENV_PROXY_ADDRESS"
-    elif [ -n "$HTTPS_PROXY" ]; then
-        ENV_PROXY_ADDRESS="$HTTPS_PROXY"
-        printf "The installer will use the proxy at '%s' (imported from env var 'HTTPS_PROXY')\n" "$ENV_PROXY_ADDRESS"
+        ENV_HTTP_PROXY_ADDRESS="$http_proxy"
+        ENV_HTTPS_PROXY_ADDRESS="$http_proxy"
+        export HTTP_PROXY="$http_proxy"
+    fi
+
+    # if https proxy is explicitly set, it overrides the inherit http proxy
+    # configuration. we also make sure we have both environment variables
+    # (with upper and lower case) set to the same value.
+    if [ -n "$HTTPS_PROXY" ]; then
+        ENV_HTTPS_PROXY_ADDRESS="$HTTPS_PROXY"
+        export https_proxy="$HTTPS_PROXY"
     elif [ -n "$https_proxy" ]; then
-        ENV_PROXY_ADDRESS="$https_proxy"
-        printf "The installer will use the proxy at '%s' (imported from env var 'https_proxy')\n" "$ENV_PROXY_ADDRESS"
+        ENV_HTTPS_PROXY_ADDRESS="$https_proxy"
+        export HTTPS_PROXY="$https_proxy"
     fi
```

Comment:
you could export https_proxy as it did before like so

```suggestion
    fi

    if [ -n "$ENV_HTTPS_PROXY_ADDRESS" ]; then
        export https_proxy="$ENV_HTTPS_PROXY_ADDRESS"
    fi
```

Reply:
Not sure if I follow, the idea here is to sanitize the inputs and guarantee that both lower and upper case versions of the variable point to the same value. `ENV_HTTPS_PROXY_ADDRESS` is read from `https_proxy` or `HTTPS_PROXY` so it is set both also are.